python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -719,12 +719,24 @@ def _process_auto_materialize_tick_generator(
 
             if sensor:
                 selection = check.not_none(sensor.asset_selection)
+                repository_origin = check.not_none(repository).get_remote_origin()
                 # resolve the selection against just the assets in the sensor's repository
                 repo_asset_graph = check.not_none(repository).asset_graph
-                eligible_keys = selection.resolve(repo_asset_graph) | selection.resolve_checks(
+                resolved_keys = selection.resolve(repo_asset_graph) | selection.resolve_checks(
                     repo_asset_graph
                 )
                 eligibility_graph = repo_asset_graph
+
+                # Ensure that if there are two asset keys defined in the same code location with automation
+                # conditions, only one of them actually launches runs from the daemon
+                eligible_keys = {
```

Comment:
one question i have here is whether we need to worry about this incorrectly filtering out observable source assets that are being auto-observed using DA - that is not a case where you would ever have the same asset defined as materializable in a different code location right?

Reply:
So specifically this would be the case where you have an auto-observed asset in one code location that is materializable in another?

I don't think that this is a thing we support at the moment -- like if you clicked on that asset in the UI and tried to execute it I think it'd break, although I'm not sure anyone has actually tried that before..

Reply:
OK yeah that was my guess as well, i would think other things later in DA would break in that case too since we are using the workspace asset graph later on